tests/lib/Mail/Provider/ManagerTest.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace Test\Mail\Provider;
+
+use OC\AppFramework\Bootstrap\Coordinator;
+use OC\AppFramework\Bootstrap\RegistrationContext;
+use OC\AppFramework\Bootstrap\ServiceRegistration;
+use OC\Mail\Provider\Manager;
+use OCP\Mail\Provider\Address;
+use OCP\Mail\Provider\IProvider;
+use OCP\Mail\Provider\IService;
+use Psr\Container\ContainerInterface;
+use Psr\Log\LoggerInterface;
+use Test\TestCase;
+
+class ManagerTest extends TestCase {
+
+	/** @var CoordinatorMockObject*/
+	private $coordinator;
```

Comment:
```suggestion
	private Coordinator&MockObject $coordinator;
```

I usually omit &MockObject because PHPStorm still provides me the proper code completion for the mock object but that's your call. However `CoordinatorMockObject` does not work ;)

The same for $container, $logger, $provider and $service.

Reply:
I must have accidentally clicked resolve on the wrong comment!